File: Python/shared/plot_FRF_r_i.py
import matplotlib.pyplot as plt

def plot_FRF_r_i(f_vec,H_vec,ax_r=None,ax_i=None,f_label='$f$ (Hz)',H_subtitle='H',lineLabel='',DispMagLines=False):
    import numpy as np

    if ax_r is None:
        ax_r=plt.subplot(2,1,1)
        if ax_i is None:
            plt.setp(ax_r.get_xticklabels(), visible=False)
            
    if ax_i is None:
        ax_i=plt.subplot(2,1,2, sharex=ax_r)

    n_f=f_vec.size
    np.reshape(f_vec,n_f)
    np.reshape(H_vec,n_f)

    ind1=f_label.find('$')
    if ind1==-1:
        raise NameError('f_label does not include LaTeX inline equation !!')
    ind2=f_label.find('$',ind1+1)
    if ind2==-1:
        raise NameError('f_label does not include LaTeX inline equation !!')
    H_subtitle1=r'%s\left(%s\right)' %(H_subtitle ,f_label[ind1+1:ind2])

    h1=ax_r.plot(f_vec,H_vec.real,label=r'%s' %lineLabel)
    ax_r.set_ylabel(r'$\Re\left(%s\right)$' %H_subtitle1)
    ax_r.grid()
    
    h2=ax_i.plot(f_vec,H_vec.imag)
    ax_i.set_xlabel(r'%s' %f_label)
    ax_i.set_ylabel(r'$\Im\left(%s\right)$' %H_subtitle1)
    ax_i.grid()
    
    return ax_r,ax_i,h1,h2

File: Matlab/shared/plot_FRF_r_i.m
function [ax_r,ax_i,h1,h2]=plot_FRF_r_i(f_vec,H_vec, ...
                                        ax_r,ax_i,f_label,H_label)   %Optional arguments

if nargin<3
    ax_r=subplot(2,1,1);
    ax_i=subplot(2,1,2);
end

if nargin<5
    f_label='$f$ (Hz)';
else
    if isempty(f_label)
        f_label='$f$ (Hz)';
    end
end
indices=strfind(f_label,'$');
if length(indices)<2,error('f_label does not include LaTeX inline equation !!'),end

H_real_multiplier='';
if nargin<6
    H_Latex_subtitle='H';
else
    if isempty(H_label)
        H_Latex_subtitle='H';
    elseif iscellstr(H_label)
        if length(H_label)~=2
            error('If H_label is cell string, it must have two elements; one for H_Latex_subtitle and the other for H_real_multiplier')
        end
        H_Latex_subtitle=H_label{1};
        H_real_multiplier=H_label{2};
    else
        H_Latex_subtitle=H_label;
    end
end
H_Latex_subtitle=[H_Latex_subtitle,'\left(',f_label(indices(1)+1:indices(2)-1),'\right)'];

h1=plot(ax_r,f_vec,real(H_vec));
%xlabel(ax_r,f_label, 'interpreter', 'latex');
%set(ax_r,'XTick',[]);
set(ax_r,'XTickLabel',[]);
ylabel(ax_r,['$\Re\left(',H_Latex_subtitle,'\right)',H_real_multiplier,'$'], 'interpreter', 'latex')
set(ax_r,'xGrid','on','YGrid','on');

h2=plot(ax_i,f_vec,imag(H_vec));
xlabel(ax_i,f_label, 'interpreter', 'latex');
ylabel(ax_i,['$\Im\left(',H_Latex_subtitle,'\right)',H_real_multiplier,'$'], 'interpreter', 'latex')
set(ax_i,'xGrid','on','YGrid','on');